### PR TITLE
CATL-1349: Display linked cases in a new tab

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview
This PR displays the linked cases tab when the setting for it is enabled. When the setting is disabled it displays the linked cases on the summary tab, as before.

## Before
Manage_Cases_case8004 (1)
The "linked cases" tab does not exist.

## After
gif

### Placeholder
Manage_Cases_case8004

## Technical Details

* We provided the `allowLinkedCasesPage` setting introduced in https://github.com/compucorp/uk.co.compucorp.civicase/pull/465 as a JS consant
* In `root-scope.decorator.js` we add the `ts` function to all scopes so we can use it in templates without having to inject this function every time.
* In `add-case-details-tabs.config.js` we conditionally add the linked cases tab depending on the value of `allowLinkedCasesPage`.

### CODE REVIEW

File: ang/civicase-base/constants.js [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
   module
     .constant('allowCaseLocks', configuration.allowCaseLocks)
+    .constant('allowLinkedCasesPage', configuration.allowLinkedCasesPage)
```
Comment (deb1990) [inline ang/civicase-base/constants.js:6]: Should not it be `allowLinkedCasesTab`?

File: ang/civicase/case/details/linked-cases-tab/directives/case-details-linked-cases-tab.directive.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+(function (angular, loadCrmForm, getCrmUrl) {
```
Comment (deb1990) [inline ang/civicase/case/details/linked-cases-tab/directives/case-details-linked-cases-tab.directive.js:1]: I think `loadCrmForm, getCrmUrl` is not being used, please check.

File: ang/test/civicase/case/details/directives/case-details.directive.spec.js [hunk 3]
```diff
@@ -374,6 +374,41 @@
         .returnValue($q.resolve({ values: CasesData.get() }));
     }));
 
+    describe('linked cases', () => {
+      let caseItem;
+
+      beforeEach(() => {
+        caseItem = _.cloneDeep(CasesData.get().values[0]);
+      });
+      describe('when the linked cases are displayed on a tab', () => {
```
Comment (deb1990) [inline ang/test/civicase/case/details/directives/case-details.directive.spec.js:383]: Empty line missing.

File: ang/civicase/case/details/directives/case-details.directive.js [hunk 4]
```diff
@@ -200,9 +202,8 @@
         delete ($scope.item.tag_id);
         _.assign($scope.item, formatCaseDetails(data));
         $scope.allowedCaseStatuses = getAllowedCaseStatuses($scope.item.definition);
-
-        $scope.$broadcast('updateCaseData');
```
Comment (deb1990) [inline ang/civicase/case/details/directives/case-details.directive.js:204]: Why are we removing these lines?

File: ang/civicase/case/details/linked-cases-tab/directives/case-details-linked-cases-tab.directive.js (hunk 2 above)
Comment (reneolivo) [reply ang/civicase/case/details/linked-cases-tab/directives/case-details-linked-cases-tab.directive.js:1]: Yes, I created this directive. These functions are being used here:

https://github.com/compucorp/uk.co.compucorp.civicase/pull/467/files#diff-3a285c51bfed1e978fe91c81d6e97996R33

File: ang/civicase/case/details/directives/case-details.directive.js (hunk 4 above)
Comment (reneolivo) [reply ang/civicase/case/details/directives/case-details.directive.js:204]: Thanks for spotting this @deb1990 ! this was going to be a regression, it was not meant to be included. I probably replaced it when trying to move the line around. Strange thing is that there are no tests for these events, so tests were passing fine. I don't have enough time to add these tests, but will keep them in the radar for the future.

**Pull request merged**